Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -843,6 +843,17 @@ private extension PurchasesOrchestrator {
                  completion: completion)
     }
 
+    func finish(_ transaction: StoreTransaction) async {
+        if let sk1Transaction = transaction.sk1Transaction {
+            self.storeKitWrapper.finishTransaction(sk1Transaction)
+        } else if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *),
+                  let sk2Transaction = transaction.sk2Transaction {
+            await sk2Transaction.finish()
+        } else {
+            // TODO: warning
+        }
+    }
```

Comment:
cc @aboedo since this came up in [our conversation](https://github.com/RevenueCat/purchases-ios/pull/1666#discussion_r892450066).

This looks nice in principle, but because `StoreKit.Transaction.finish` is `async`, we'd have to make a lot of `PurchasesOrchestrator` `async` just to support this.

I wanted to open this PR just in case what you had in mind was different. But I think what's best is what we have right now: finishing transactions in 2 different places for SK1 and SK2.

Reply:
we don't really need to wait for the result, though, right? Since we're not doing anything with it? 
i.e.: wouldn't we be able to have this function _not_ be async, and then just have it start a task that calls `finish` for sk2?

Reply:
I thought about it. There could be race conditions, but that's unclear not knowing exactly what finish does. The docs don't make it clear,, but I suspect you're probably right!